superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,202 @@
+                        self._name, op, ' '.join(list(self.__operations.keys()))
+                    )
+                )
+                return False
+            else:
+                self._bin_name = self.__operations[op]
+                if not self._set_binary_path():
+                    return False
+
+                command = os.path.join(self._args.bin_dir, self._bin_name)
+                command += ' -b {} -e {} -f {} -g {} -c {}'.format(
+                    self._args.b, self._args.e, str(self._args.f), str(self._args.g), str(self._args.c)
+                )
+                self._commands.append(command)
+
+        return True
+
+    def _process_raw_result(self, cmd_idx, raw_output):    # noqa: C901
+        """Function to parse raw results and save the summarized results.
+
+          self._result.add_raw_data() and self._result.add_result() need to be called to save the results.
+
+        Args:
+            cmd_idx (int): the index of command corresponding with the raw_output.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        # If it's invoked by MPI and rank is not 0, empty content is expected
+        if os.getenv('OMPI_COMM_WORLD_RANK'):
+            rank = int(os.getenv('OMPI_COMM_WORLD_RANK'))
+            if rank > 0:
+                return True
+
+        self._result.add_raw_data('raw_output_' + self._args.operations[cmd_idx], raw_output)
+
+        content = raw_output.splitlines()
+        size = -1
+        busbw_out = -1
+        time_out = -1
+        algbw_out = -1
+        try:
+            # Filter useless output
+            out_of_place_index = -1
+            out_of_bound_index = -1
+            for index, line in enumerate(content):
+                if 'out-of-place' in line:
+                    out_of_place_index = index
+                if 'Out of bounds values' in line:
+                    out_of_bound_index = index
+            content = content[out_of_place_index + 1:out_of_bound_index]
+            # Parse max out of bound bus bw as the result
+            size_index = -1
+            time_index = -1
+            busbw_index = -1
+            algbw_index = -1
+            for line in content:
+                if 'time' in line and 'busbw' in line:
+                    # Get index of selected column
+                    line = line[1:].strip(' ')
+                    line = re.sub(r' +', ' ', line).split(' ')
+                    # Get first index of condition or default value in list
+                    size_index = next((i for i, x in enumerate(line) if x == 'size'), -1)
+                    time_index = next((i for i, x in enumerate(line) if x == 'time'), -1)
+                    busbw_index = next((i for i, x in enumerate(line) if x == 'busbw'), -1)
```

Comment:
maybe you need `line.index('busbw') - len(line)` to count from end here

Reply:
changed